src/iminuit/typing.py

```diff
@@ -39,3 +34,17 @@ class LossFunction(Protocol):
     def __call__(self, z: NDArray) -> NDArray:
         """Evaluate loss function on values."""
         ...  # pragma: no cover
+
+
+class ValueRange(NamedTuple):
```

Comment:
Personally, I'd make these Protocols, not NamedTuples. NamedTuple is a backward compatibility feature with a lot of bloat (tuple methods, for example). But don't think it really matters for this.

Reply:
Oops, forgot they need to be instantiated. 🤦 Then would go with a dataclass over NamedTuple. Again, not at all a problem.

Reply:
Ok, I will use dataclass, I was not aware that NamedTuple from typing carries that much bloat and made this choice not thinking much about it. A dataclass is more idiomatic modern Python, I agree.

Reply:
I will keep this convo unresolved as a reminder to self.

Reply:
Actually, NamedTuple was correct, because I wanted tuple semantics and a dataclass does not give you that.